pkg/api/handlers/compat/images_push.go

```diff
@@ -76,7 +75,11 @@ func PushImage(w http.ResponseWriter, r *http.Request) {
 	if _, found := r.URL.Query()["tlsVerify"]; found {
 		options.SkipTLSVerify = types.NewOptionalBool(!query.TLSVerify)
 	}
-	if err := imageEngine.Push(context.Background(), imageName, query.Destination, options); err != nil {
+	destination := query.Destination
+	if destination == "" {
```

Comment:
```suggestion
	if _, found := r.URL.Query()["destination"]; !found {
```
Would be more correct if looking for a missing parameter.

Reply:
@jwhonce fixed